### DISCUSSION TITLE
Prompt image upload

### DISCUSSION THREAD

**Discussion by suhsin85** [beeware/toga#2784] [Q&A]
How do you prompt for image upload from user of the app from the hard drive? Is such a function available on Toga?
I would like to make an app where it takes an image from either an iphone or an android, reads the metadata of the image, and gives feedback on the metadata.

**Comment by freakboy3742** [comment]
Toga provides a [file selection dialog](https://toga.readthedocs.io/en/stable/reference/api/window.html#toga.Window.open_file_dialog); however, on mobile, this dialog isn't implemented because phones don't expose a traditional "filesystem" in the same way that a desktop machine does. 

#1158 contains an old prototype version of this feature on Android; however, the objects you get back from this dialog aren't Python file objects - they're wrappers around Java's Storage Access Framework.  A conceptually similar approach would be needed on iOS, but the implantation would be radically different.

#1171 is an open ticket for a cross-platform file API that would present a more traditional Python file API around the SAF, as well as other file wrappers.

**Reply by suhsin85** [reply]
Thank you. Do you know if there is any other Python module that would allow me to prompt photos/files upload from android and iOS?

**Reply by freakboy3742** [reply]
Not really - iOS and Android support for Python is relatively new, so there's very few GUI libraries for wrapping those capabilities.
